apis/tempo/v1alpha1/microservices_webhook.go

```diff
@@ -25,8 +27,9 @@ var _ webhook.Defaulter = &Microservices{}
 // Default implements webhook.Defaulter so a webhook will be registered for the type.
 func (r *Microservices) Default() {
 	microserviceslog.Info("default", "name", r.Name)
-
-	// TODO(user): fill in your defaulting logic.
+	if r.Spec.Retention.Global.Traces == 0 {
```

Comment:
can be retention set to 0? In some systems it means unlimited retention. If that is the case we should change `Traces` to a pointer.

Reply:
They dont consider 0 at all.
https://github.com/grafana/tempo/blob/9c296831384c29dc9b706601b1580b83b77b9d90/modules/compactor/config.go#L36
https://github.com/grafana/tempo/blob/9c296831384c29dc9b706601b1580b83b77b9d90/pkg/util/config.go#L70-L76